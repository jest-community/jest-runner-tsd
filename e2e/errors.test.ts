import runJest from './runJest';

test('throws if syntax error is encountered', async () => {
  expect(await runJest('errors-syntax')).toMatchSnapshot();
});

test('throws if nearest tsconfig.json is not valid', async () => {
  expect(await runJest('errors-tsconfig')).toMatchSnapshot();
});
