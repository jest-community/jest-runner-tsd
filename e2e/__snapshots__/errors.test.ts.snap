// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws if nearest tsconfig.json is not valid 1`] = `
"FAIL e2e/__fixtures__/errors-tsconfig/index.test.ts
  ● Test suite failed to run
    ConfigError: Compiler option 'strict' requires a value of type boolean.
      3 |     \\"module\\": \\"commonjs\\",
      4 |     \\"target\\": \\"es2019\\",
    > 5 |     \\"strict\\": \\"yes\\",
        |               ^
      6 |     \\"noEmitOnError\\": true
      7 |   },
      8 |   \\"include\\": [\\"./**/*\\"]
      at tsconfig.json:5:15
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:
Ran all test suites.
"
`;

exports[`throws if syntax error is encountered 1`] = `
"FAIL e2e/__fixtures__/errors-syntax/index.test.ts
  ● Test suite failed to run
    SyntaxError: ')' expected.
      2 | import one from '.';
      3 |
    > 4 | expectError(one('foo', 'bar');
        |                              ^
      5 | expectError(one('foo' 'bar'));
      6 |
      at index.test.ts:4:30
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:
Ran all test suites.
"
`;
